### PR TITLE
Preparations for more testchains

### PR DESCRIPTION
**Pull request by jtimon**
Separated from #8994 as suggested by MarcoFalke and Sjors in https://github.com/bitcoin/bitcoin/pull/8994#issuecomment-522555390

You can't really test the qt changes on their own, so to test them, use #8994 .

### CODE REVIEW

File: src/chainparamsbase.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@ const std::string CBaseChainParams::REGTEST = "regtest";
 
 void SetupChainParamsBaseOptions()
 {
+    gArgs.AddArg("-chain=<chain>", "Use the chain <chain> (default: main). Allowed values: main, testnet, regtest", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
```
Comment (MarcoFalke) [inline src/chainparamsbase.cpp]: I don't think `testnet` is an allowed value

```suggestion
    gArgs.AddArg("-chain=<chain>", "Use the chain <chain> (default: main). Allowed values: main, test, regtest", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
```

Alternatively, you can use 
```cpp
Join({
  CBaseChainParams::MAIN,
  CBaseChainParams::TESTNET,
  CBaseChainParams::REGTEST,
}, ", ");

**Comment by jtimon** [comment]
Fixed @MarcoFalke 's nit and slightly improve documentation.

**Comment by Sjors** [comment]
ACK dba00d2

It was fairly easy to cherry-pick these commits onto the signet PR (https://github.com/Sjors/bitcoin/commit/df1846c8ae97f28165c0d0b44c6cd7ae6b1fcbba & https://github.com/bitcoin/bitcoin/commit/124ee3786802b19e81351808f3145ed4e80dc32c, cc @kallewoof). By dropping the `signet` lines from the QT commit, I indeed get a purple client.

**Comment by kallewoof** [comment]
Nice!

**Comment by jtimon** [comment]
@Sjors sure, that's another way to test it, thanks.

File: src/qt/networkstyle.cpp [hunk 2]
```diff
@@ -85,8 +88,8 @@ const NetworkStyle *NetworkStyle::instantiate(const QString &networkId)
                     network_styles[x].appName,
                     network_styles[x].iconColorHueShift,
                     network_styles[x].iconColorSaturationReduction,
-                    network_styles[x].titleAddText);
+                    titleAddText.c_str());
         }
     }
-    return nullptr;
+    return new NetworkStyle(strprintf("%s-%s", QAPP_APP_NAME_CUSTOM, networkId).c_str(), 250, 30, titleAddText.c_str());
```
Comment (MarcoFalke) [inline src/qt/networkstyle.cpp]: in commit dba00d2b04c3bf0fd4d31cca662d262c459f2ebb:

This is dead code and I'd prefer if the "default purple" and "custom chain" concept were left for the testchains pull request. This is not required for signet (apart from adding dead code).

I am fine with the `titleAddText` change in this commit, though. So you can either remove the commit or strip it of the concept of a custom chain.

**Review by maflcko — DISMISSED** [review]
ACK 36f9573e82d370af76fed0789fe3d671546e6e85 (also run the test to verify that only error messages are responsible for change the hash)

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 36f9573e82d370af76fed0789fe3d671546e6e85 (also run the test to verify that only error messages are responsible for change the hash)
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUga+Av/Y3yW/KQZwNdF7hYFnrmx2Oi6tYorQ3duhA1xTtkkNMZJn6VHS2qDDoOL
8OApQqJa8AQOQO+Ds1kkXQMq7PJzxaxh0kw31YvmQblZGKPsUUqziCbHF1GjFHWo
2GhY3x+sBuNhHyTXgS6EqiyEmEAE0B+1fb4xvAC7cna5RkexOR4JDu6n6I5PZDxu
5upGAB55q1014wUc1aNbEqFHwDmqiwAzIMIKgWSpaQoFKNOUsNl0MS/XpSTeNG7u
54LHuSoTyLJi+luhY0I+wqDMZuzgJeSZySlQFKDMzC0pVLoMYfO0qqcpDD4pkSkW
cg55tpewD4W/ljq4lFYTATtHzAy6mi3s/AN09TM/GE7DOyYdnSxPRyBrKaA039Ii
HCTUZtS1hv/gbs7l6aqUgWlNFwicSvqRSHWJ7IXmbqIS26E5R+e9i4II6mfcZsoD
GQXjYagSkdta8xydIaWDTHt6x2hXzYHXxtUyI6reJ+ObrCgsPjiWG6eA3aUoaaYv
iEIzcc2c
=ubtk
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `fe315895de9bcfebf0a22dee8eb70c8cd08dd7ba0736e8d0c931542ea93a4bd7  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108fe315895de9bcfebf0a22dee8eb70c8cd08dd7ba0736e8d0c931542ea93a4bd7f010558cd9d50b88453c81811f4a3551fbb808fff0105c8929cd7eca42a98493ad4812194bf708f120dc386091f791eaadbfa1debe1998201e01ec1a69fcc5352a8e4d100fea9e996b08f12082c2b5bc28bcebd7daddb7dbc12aa8eb796d61bb0261db20f1157052ef01ff7608f120d2136f852371012c608040922db5dbca27b13bca9d998560042d52e6dfe7d46808f120494e1ef5993f5b2a8dec5d40c1da7b19609a686ba8146778bb9d73a702f0e3ea08f0202327e35b2b8dc0ed5d03a3e2307e0338ef9b996a6a1e549d311c48011bab760d08f020b8626f3e414dd4603b26dfecef9a2000be44316d3748ac14c94f35d1daf563fe08f1045d727248f00825c979630a25bfd50083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0107b0eaa1d7ee6aac20b13abd92c439fe908f120725361b6d930bfc3b08139963a4852868ffd83b44682cce64a8d5394a8bfc0d608f1206f8f705d1872897aa21bc6dcee0bca5bdfba3368edf213e2ffff394f45a31bba08f12082f79648b014a0f0ea71f251c98a84d4318b68acc7b9529572098f20176af47108f120297fcc9100e172e6c82b16bcaca7f7f5e240bcc3d7e2a56fa01ce3b11636eb2a08f120e5055f1fc2fabba09b02d35bbd8494657a5eeee3aa5b41c567156c165d1548db08f02001bdf9f25c6ef9666d37f7865f140f8cfefc6118fc8e1778f102a449e607225408f1045d727248f0087b6feed4be19f5f40083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff01086ac270af744d8c8d5bdd2d5c9d495ec08f12042a1982cb9afc812b8982896a30a34a81361a38690856f44c1314cd90242d61308f020370608f36b449fcd2d8e9faff9682ffd31a14c39f07f3e821977958fe9f4f38408f120aec0e2a0475b77717c0afc069dc1e37be25c71477a48f2614aae861383b2e6b108f020fa4ada593c157f03cb69beff9e7e7448a86191a8610f5b8215a715fd4634fe9c08f020a3da086e7b3f7cbd47120da9dd16b662b812169da5a147ee63e12b5cc6bf078308f02057d3da95c629a8063b15172d55b084e786348abd4f94ba9fe5f5a75a660d8d9a08f1045d727248f008f25525dc591561490083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010afbfbe566b2858997e31aaa45753661a08f1203add123e8cfe16911577c30df3d9e3a32bdbc66c2997d9a27292e3d6b6edd3a308f1203007fe594c8e2300d2bdcb71704c2de4c617251c0658aa3432c646359db41dca08f020a1ce1cfafa3095a05379a08765e20624300025d1faf493b49cd2cdcf7540ed9908f1208a0172f68ee9b38c1078024f4e92dd8e3710689b3f3d5f3f92a788d30f34bbdd08f0204b3636f18559ea096dcb52ead880ad85fa60019fbe866c7ced28ae1b16b9f8d108f120b0918e33ff04779d89c0590d7e0ad20aa62dd9825d29e42286f8b5a9b004cb9b08f1045d727248f0083b061ef26db4df070083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>

File: src/qt/networkstyle.cpp (hunk 2 above)
Comment (jtimon) [reply src/qt/networkstyle.cpp]: I guess my idea was for signet to reuse this color, perhaps change the tittle from custom to signet if it comes first, and if not just reuse the custom name for all custom chains signet or not.

Comment (MarcoFalke) [reply src/qt/networkstyle.cpp]: Wouldn't that also require 2f5796504a5d984276931da974436a8a5d352f74?

Comment (jtimon) [reply src/qt/networkstyle.cpp]: hopefully fixed

**Comment by MarcoFalke** [comment]
ACK 3bf9d8cac09fc88727ef2f2a2bea33b90b625e50

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 3bf9d8cac09fc88727ef2f2a2bea33b90b625e50
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhhSgv/d0tNdJAcYe5izbrwgb32bUYpXXLrk7KdCxauZoGCVKsPxbK+iGx5nhZg
OcUun64mGXA9L6fMLFyByXZe1FND1CVNHF/b/5mxobmpaE5RqFq7DWhpcVNoXwe3
7CukYbaPMcqajlIsr1KHgetHQrs5Z7HQO4qC4+vvcVLTBd+X0LB61AKbEaDwVldp
oFKk4C7B9wNDfnUxGqjyBr671tfMgGVi+rDlmra4xedUd93PTRJ12fudf1pp9OcI
hvr8ablChiTG1EKbbfbXPcScpfsfjGc1ZI0z2wvKXKRSpe1pmoWLje9ySt0TUBN9
B8wRdozzk4Q63v+6X2+JvNzT7gIEJYi7wNgERipkTJAckWgwZ+jU3WgcB+yHkfrf
Kun/DULfIusSIqSJBKU8UKe3zCxH2ykhRJT3tzd7zB5k+QHGgWLq9ojLpBdfi/jm
hkhxq1TIqrMmwC3jXWnTH4P1Qtvx5Fiq4KB7RmXjuZ43EIIHMlet8gewhZRBH0X+
DXnwuC7C
=omI2
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `9eb4317ea08a354e0ed72ca0faed34267277bab19a079787f864956aa279387d  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401089eb4317ea08a354e0ed72ca0faed34267277bab19a079787f864956aa279387df0100cc8334ff31a7bd8574046f399b165aa08fff0101873b978f03c6e9f89a0409f7cd69c9b08f1045d736250f008e1d4dd038dc98d960083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0108b0fa317ecb71d3980c7345373ad9ccb08f1045d736250f008f724758d7f6c515b0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010c2cf533dc7dc6499ae2cc394879f23ac08f1045d736251f00896af07a103c3a5e70083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Pull request merged**